conda_smithy/lint_recipe.py

```diff
@@ -490,18 +486,17 @@ def lintify_meta_yaml(
                         lints.append(
                             "`noarch` packages can't have selectors. If "
                             "the selectors are necessary, please remove "
-                            "`noarch: {}`.".format(noarch_value)
+                            f"`noarch: {noarch_value}`."
                         )
                         break
 
     # 19: check version
-    if package_section.get("version") is not None:
-        ver = str(package_section.get("version"))
+    if (version := package_section.get("version")) is not None:
         try:
-            VersionOrder(ver)
-        except:
+            VersionOrder(str(version))
+        except InvalidVersionSpec as e:
             lints.append(
-                "Package version {} doesn't match conda spec".format(ver)
+                f"Package version {version} doesn't match conda spec: {e}"
             )
 
```

Comment:
@h-vetinari This needed to be changed because the except clause is too broad.

Therefore I catch the only error that can happen here, `InvalidVersionSpec`, and, in one process, also print the `InvalidVersionSpec` error details because it helps debugging.